src/main/Extensions/SimpleFolderSearch/SimpleFolderSearch.ts

```diff
@@ -0,0 +1,121 @@
+import { AssetPathResolver } from "@Core/AssetPathResolver";
+import { Extension } from "@Core/Extension";
+import { FileSystemUtility } from "@Core/FileSystemUtility";
+import { SettingsManager } from "@Core/SettingsManager";
+import { Translator } from "@Core/Translator";
+import { App } from "electron";
+import path from "path";
+
+export class SimpleFolderSearch implements Extension {
+    public readonly id = "SimpleFolderSearch";
+    public readonly name = "Simple Folder Search";
+
+    public readonly author = {
+        name: "Sasha Weiss",
+        githubUserName: "sashaweiss",
+    };
+
+    private readonly defaultSettings: Record<OperatingSystem, { folders: string[] }> = {
+        Windows: {
+            folders: [this.app.getPath("home")],
+        },
+        macOS: {
+            folders: [this.app.getPath("home")],
+        },
+        Linux: {
+            folders: [this.app.getPath("home")],
+        },
+    };
+
+    private readonly folderIconImage: Image = {
+        url: `file://${this.assetPathResolver.getExtensionAssetPath(this.id, "macos-folder-icon.png")}`,
+    };
+
+    public constructor(
+        private readonly app: App,
+        private readonly operatingSystem: OperatingSystem,
+        private readonly assetPathResolver: AssetPathResolver,
+        private readonly fileSystemUtility: FileSystemUtility,
+        private readonly settingsManager: SettingsManager,
+        private readonly translator: Translator,
+    ) {}
+
+    public async getSearchResultItems(): Promise<SearchResultItem[]> {
+        type SearchResult = {
+            fullPath: string;
+            abbreviatedName: string;
+            fileName: string;
+        };
+
+        const searchResultBatches: SearchResult[][] = await Promise.all(
+            this.getPathsToSearch().map(async (folderPath: string): Promise<SearchResult[]> => {
+                const directoryEntries = await this.fileSystemUtility.readDirectory(folderPath);
+                return directoryEntries.map((directoryEntry): SearchResult => {
+                    return {
+                        fullPath: directoryEntry,
+                        abbreviatedName: directoryEntry.replace(this.app.getPath("home"), "~"),
+                        fileName: path.parse(directoryEntry).base,
+                    };
+                });
+            }),
+        );
+
+        const searchResults: SearchResult[] = searchResultBatches.reduce(
+            (accumulator: SearchResult[], nextValue: SearchResult[]): SearchResult[] => {
+                return accumulator.concat(nextValue);
+            },
+            [],
+        );
+
+        // Filter out non-directories, and hidden items.
+        const filteredSearchResults: SearchResult[] = searchResults.filter((searchResult: SearchResult): boolean => {
+            return this.fileSystemUtility.isDirectory(searchResult.fullPath) && !searchResult.fileName.startsWith(".");
+        });
```

Comment:
Could make this configurable, although these are the settings I prefer and I figured for a V1 they were reasonable defaults.